### DISCUSSION TITLE
Using parameters through components.

### DISCUSSION THREAD

**Discussion by alexcmartins** [Budibase/budibase#9598] [Help] [answered]
Hello everybody.

I have some API endpoints, but in some cases I need to pass a parameter, the problem is that the structure of the API url is different from what is requested.
Eg: ?=

Issa is the URL I need to call.
![image](https://user-images.githubusercontent.com/61984606/217675837-936d8789-20bd-4fd0-a607-e3c84957a17c.png)

The way I'm trying to do.
![image](https://user-images.githubusercontent.com/61984606/217676060-bfe8134c-8613-48da-9063-acb8e57fce97.png)

I already tried using Binding and Params, but then the URL adds ? and =, so it doesn't return the answer.
Can anyone help me?
Thanks!

**Comment by melohagan** [comment, marked as answer]
Hey @alexcmartins 

If Budibase is adding *?* to the URL, then you probably have **params** added accidentally? Check the **Params** tab and make sure nothing is there.

**Reply by alexcmartins** [reply]
Hello @melohagan 
Thanks for your feedback, being more objective what I need is to get the data from a field of an element and insert it into my url for the query, I tried just passing the binding but I couldn't and I also tried passing a parameter.
What should I do?
Thanks again!

Best regards.

**Reply by melohagan** [reply]
1. Add a REST query binding
![Screenshot 2023-02-14 at 16 02 38](https://user-images.githubusercontent.com/101575380/218791560-04866dbd-f4d7-4ec2-a86c-3868e98a1027.png)

2. Add a param using the binding
![Screenshot 2023-02-14 at 16 03 44](https://user-images.githubusercontent.com/101575380/218791815-9abff907-7107-44fe-9d66-d32048ebed4c.png)

3. Save your query

4. In the **Design** section, add a *Data provider* and select your REST query
![Screenshot 2023-02-14 at 16 04 46](https://user-images.githubusercontent.com/101575380/218792063-eb1a4609-e8cc-403f-90e1-e9ffbe4afdb8.png)

5. Press the cog icon. Pass in a binding value -> it could be a static value or a binding such as a form field:
![Screenshot 2023-02-14 at 16 07 09](https://user-images.githubusercontent.com/101575380/218792676-3683cca7-d01c-4716-8a8e-05f29cd99401.png)

If you want to make a POST request, an example can be found here: https://docs.budibase.com/docs/custom-queries#tutorial-posting-form-data-to-a-custom-query
*(It's using a Custom SQL query, but the principle would be the same for a POST API call)*

**Reply by alexcmartins** [reply]
@melohagan Perfect, I did it right.
The question is, my endpoint doesn't accept ?q=, is there a way to get the value of an element and pass it to a URL without using the binding that way?
![example](https://user-images.githubusercontent.com/61984606/218796626-01e869d3-786d-4f8c-84ce-97bc83107180.png)

I have to go this way...
![example_1](https://user-images.githubusercontent.com/61984606/218796880-ff2049bd-256e-42d6-b46a-eb2221c12aa6.png)

Best regards.

**Reply by melohagan** [reply]
Absolutely. In your case then you would simply not include the params, but you still need the bindings in your REST query.

Then in the URL you would manually type the binding syntax, for example: `myendpoint.com/{{ MyBinding }}`

![Screenshot 2023-02-14 at 16 27 46](https://user-images.githubusercontent.com/101575380/218797820-1fbe9e94-c7ea-4373-aba6-455f5d2dc11b.png)

**Reply by alexcmartins** [reply]
@melohagan 
I really appreciate the help, the API is currently under maintenance, but I'm going to test it with another one I have, because it works the same way!

I will add the images below to verify that I followed your steps correctly.
![example_3](https://user-images.githubusercontent.com/61984606/218800972-5bdb18d2-7385-4074-8743-6b0dc615ec43.png)
![example_2](https://user-images.githubusercontent.com/61984606/218801006-7275da7b-a4de-4f22-9bed-ebbf592b20da.png)

Thank you!

**Reply by melohagan** [reply]
That looks right to me!
